machineconfiguration/v1alpha1/types_internalreleaseimage.go

```diff
@@ -0,0 +1,145 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=internalreleaseimages,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2510
+// +openshift:file-pattern=cvoRunLevel=0000_80,operatorName=machine-config,operatorOrdering=01
+// +openshift:enable:FeatureGate=NoRegistryClusterOperations
+// +kubebuilder:metadata:labels=openshift.io/operator-managed=
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="internalreleaseimage is a singleton, .metadata.name must be 'cluster'"
+
+// InternalReleaseImage is used to keep track and manage a set
+// of release bundles (OCP and OLM operators images) that are stored
+// into the control planes nodes.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type InternalReleaseImage struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the configuration of this internal release image.
+	// +required
+	Spec InternalReleaseImageSpec `json:"spec,omitzero"`
```

Comment:
I think this should also have `omitempty` per https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#serialization-of-optionalrequired-fields ? Although I'm not 100% sure on this

Reply:
omitempty wouldn't actually work here, though we have tended to include it for consistency, I don't have a strong opinion either way tbh